shard/sharder.go

```diff
@@ -0,0 +1,421 @@
+package shard
```

Comment:
I think since we refer to this as the Sharder we should put it in a `sharder` folder and name the package `sharder` too.